### PR TITLE
Backtrack

### PR DESCRIPTION
**Pull request by Peter9192**
This PR further refactors the code for backtracking. The whole sample case now runs on my laptop in under 10 seconds. Some of the important changes are:

* Abbreviate commonly used code (e.g. `shift_east(...)`, `[inner]` and `s_track_relative`)
* Remove as many as possible empty array allocations
* Do the daily accumulations "on the fly", i.e. don't store intermediate values (3d) and then sum them, but just add them directly to a single (2d) output array.
* Compute intermediate variables (e.g. fx_e_lower_we) only one time step at a time. That saves a lot of working memory.

Please review thoroughly to identify any mistakes I might have made.

### CODE REVIEW

File: backtrack.py [hunk 1]
```diff
@@ -31,424 +31,243 @@ def input_path(date):
...
-    states = xr.Dataset(
+    # Pack processed data into new dataset
+    ds = xr.Dataset(
         {
-            "s_track_upper": (["time_states", "lat", "lon"], s_track_upper),
-            "s_track_lower": (["time_states", "lat", "lon"], s_track_lower),
+            "s_track_upper_restart": (["lat", "lon"], s_track_upper),  # Keep last state for a restart
+            "s_track_lower_restart": (["lat", "lon"], s_track_lower),
+            "s_track_upper": (["lat", "lon"], s_track_upper_mean),
+            "s_track_lower": (["lat", "lon"], s_track_upper_mean),
```
Comment (Imme1992) [inline backtrack.py:229]:
```suggestion
            "s_track_lower": (["lat", "lon"], s_track_lower_mean),
```

**Review by Imme1992 — COMMENTED** [review]
Some small comments on naming variables etc. and one bug where s_track_lower was named s_track_upper

**Comment by Imme1992** [comment]
I think there are still negative values in e_track left as the evaporation variable itself can still be negative, preprocess script is not the same script as in Master branch (where evaporation is made always positive)

**Review by Imme1992 — APPROVED** [review]

**Comment by ruudvdent** [comment]
todo: re-check negative values after updated pre-processing from backtrack

**Pull request merged**